File: Day-2/solutions_2.py
try:
    n = int(input("Enter number of elements: "))

    print("Enter elements:")
    arr = list(map(int, input().split()))

    if len(arr) != n:
        print("Invalid input! Number of elements does not match.")
        exit()

    running_sum = []
    total = 0

    for num in arr:
        total += num
        running_sum.append(total)

    print("Running Sum:", running_sum)

except ValueError:
    print("Invalid input! Please enter integers only.")

File: Day-2/solutions_2.cpp
#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int n;
    cout << "Enter number of elements: ";
    cin >> n;

    if(cin.fail() || n <= 0)
    {
        cout << "Invalid input!" << endl;
        return 0;
    }

    vector<int> arr(n);

    cout << "Enter elements: ";
    for(int i = 0; i < n; i++)
    {
        cin >> arr[i];
        if(cin.fail())
        {
            cout << "Invalid input! Please enter integers only." << endl;
            return 0;
        }
    }

    vector<int> running_sum(n);

    running_sum[0] = arr[0];

    for(int i = 1; i < n; i++)
    {
        running_sum[i] = running_sum[i-1] + arr[i];
    }

    cout << "Running Sum: ";
    for(int i = 0; i < n; i++)
    {
        cout << running_sum[i] << " ";
    }

    return 0;
}